src/sweets/core.py

```diff
@@ -214,21 +228,43 @@ def model_construct(cls, _fields_set=None, **values):
             **values,
         )
 
-    def __init__(self, **data: Any) -> None:
-        super().__init__(**data)
-        # Track the directories that need to be created at start of workflow
-        self.log_dir = self.work_dir / "logs"
-        self.gslc_dir = self.work_dir / "gslcs"
-        self.geom_dir = self.work_dir / "geometry"
-        self.ifg_dir = self.work_dir / "interferograms"
-        self.stitched_ifg_dir = self.ifg_dir / "stitched"
-        self.unw_dir = self.ifg_dir / "unwrapped"
-        self._dem_filename = self.work_dir / "dem.tif"
-        self._water_mask_filename = self.work_dir / "watermask.flg"
-
-        # Expanded version used for internal processing
+    # Track the directories that need to be created at start of workflow
+    @computed_field  # type: ignore[prop-decorator]
+    @property
+    def log_dir(self) -> Path:
+        return self.work_dir / "logs"
+
+    @computed_field  # type: ignore[prop-decorator]
+    @property
+    def gslc_dir(self) -> Path:
+        return self.work_dir / "gslcs"
+
+    @computed_field  # type: ignore[prop-decorator]
+    @property
+    def geom_dir(self) -> Path:
+        return self.work_dir / "geometry"
+
+    @computed_field  # type: ignore[prop-decorator]
+    @property
+    def ifg_dir(self) -> Path:
+        return self.work_dir / "interferograms"
+
+    @computed_field  # type: ignore[prop-decorator]
+    @property
+    def stitched_ifg_dir(self) -> Path:
+        return self.ifg_dir / "stitched"
+
+    @computed_field  # type: ignore[prop-decorator]
+    @property
+    def unw_dir(self) -> Path:
+        return self.ifg_dir / "unwrapped"
+
+    # Expanded version used for internal processing
+    @computed_field  # type: ignore[prop-decorator]
```

Comment:
Do we want this one to be a `computed_field` if it's internal?

Reply:
Good point.
`_orbit_dir_is_set` and `_data_dir_is_set` are now exported to yaml.
I will make sure these are exempt from export as well.

Reply:
Now that i take a look at it, these are not being used, so I remove them.
Otherwise, we can use the `model_fields_set` attribute from a pydantic class.